src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,893 @@
+            {
+                loggers.Add(new BinaryLogger
+                {
+                    // Default the binlog parameters if only the debug option was specified
+                    Parameters = binlogParameters ?? "LogFile=nuget.binlog"
+                });
+            }
+
+            var projects = new ConcurrentDictionary<string, ProjectWithInnerNodes>(StringComparer.OrdinalIgnoreCase);
+
+            var projectCollection = new ProjectCollection(
+                globalProperties: null,
+                // Attach a logger for evaluation only if the Debug option is set
+                loggers: loggers,
+                remoteLoggers: null,
+                toolsetDefinitionLocations: ToolsetDefinitionLocations.Default,
+                // Having more than 1 node spins up multiple msbuild.exe instances to run builds in parallel
+                // However, these targets complete so quickly that the added overhead makes it take longer
+                maxNodeCount: 1,
+                onlyLogCriticalEvents: false,
+                // Loading projects as readonly makes parsing a little faster since comments and whitespace can be ignored
+                loadProjectsReadOnly: true);
+
+            var failedBuildSubmissions = new ConcurrentBag<BuildSubmission>();
+
+            try
+            {
+                var sw = Stopwatch.StartNew();
+
+                var evaluationContext = EvaluationContext.Create(EvaluationContext.SharingPolicy.Shared);
+
+                ProjectGraph projectGraph;
+
+                int buildCount = 0;
+
+                var buildParameters = new BuildParameters(projectCollection)
+                {
+                    // Use the same loggers as the project collection
+                    Loggers = projectCollection.Loggers,
+                    LogTaskInputs = Debug
+                };
+
+                // BeginBuild starts a queue which accepts build requests and applies the build parameters to all of them
+                BuildManager.DefaultBuildManager.BeginBuild(buildParameters);
+
+                try
+                {
+                    // Create a ProjectGraph object and pass a factory method which creates a ProjectInstance
+                    projectGraph = new ProjectGraph(entryProjects, projectCollection, (path, properties, collection) =>
+                    {
+                        var projectOptions = new ProjectOptions
+                        {
+                            EvaluationContext = evaluationContext,
+                            GlobalProperties = properties,
+                            // Ignore bad imports to maximize the chances of being able to load the project and restore
+                            LoadSettings = ProjectLoadSettings.IgnoreEmptyImports | ProjectLoadSettings.IgnoreInvalidImports | ProjectLoadSettings.IgnoreMissingImports | ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition,
```

Comment:
This is a change from current restore. 
ProjectLoadSettings.IgnoreMissingImports is not the default in current restore. 

Granted the move was for performance reasons (in 15.5), but it changed the experience for non-SDK based projects. 
There'll be a difference in the experience :)

Reply:
fwiw, I get the arguments for having it, but we try to avoid having different experiences across client, because it just adds cognitive overload for both us and customers trying to figure out how the heck things are supposed to work, or why they work in client but break in another.

Reply:
I believe its already this way for `msbuild /t:restore` and Visual Studio since they both set these flags.  So its only different from `nuget.exe restore` but we're working to deprecate that.

Reply:
I'm certain that msbuild /t:restore doesn't, because people have hit it before :)
Visual Studio won't load projects whose imports cannot be resolved for the most part.

Reply:
Here is where MSBuild sets the flags: https://github.com/microsoft/msbuild/blob/master/src/MSBuild/XMake.cs#L1356

Reply:
Maybe the effect of these is not actually the one that I am thinking of. 

I have forwarded you a thread about that, to help clarify it.

Reply:
All of my confusion is summarized in https://github.com/NuGet/Home/issues/9035. 

No regression, this change is not relevant for this PR.